server/ips.go

```diff
@@ -20,3 +24,147 @@ func GetOutboundIP() (net.IP, error) {
 
 	return localAddr.IP, nil
 }
+
+// addrToIPNet casts addr to IPNet.
+// Returns nil if addr is not of IPNet type.
+func addrToIPNet(addr net.Addr) *net.IPNet {
+	switch v := addr.(type) {
+	case *net.IPNet:
+		return v
+	default:
+		return nil
+	}
+}
+
+// FilterAddressesForPairingServer filters private unicast addresses.
+// ips is a 2-dimensional array, where each sub-array is a list of IP
+// addresses for a single network interface.
+func FilterAddressesForPairingServer(ips [][]net.IP) []net.IP {
+	var result []net.IP
+
+	for _, niIps := range ips {
+		var ipv4, ipv6 []net.IP
+
+		for _, ip := range niIps {
+
+			// Only take private global unicast addrs
+			if !ip.IsGlobalUnicast() || !ip.IsPrivate() {
+				continue
+			}
+
+			if v := ip.To4(); v != nil {
+				ipv4 = append(ipv4, ip)
+			} else {
+				ipv6 = append(ipv6, ip)
+			}
+		}
+
+		// Prefer IPv4 over IPv6 for shorter connection string
+		if len(ipv4) == 0 {
+			result = append(result, ipv6...)
+		} else {
+			result = append(result, ipv4...)
+		}
+	}
+
+	return result
+}
+
+// GetLocalAddresses returns an array of all addresses
+// of all available network interfaces.
+func GetLocalAddresses() ([][]net.IP, error) {
+	nis, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	var ips [][]net.IP
+
+	for _, ni := range nis {
+		var niIps []net.IP
+
+		addrs, err := ni.Addrs()
+		if err != nil {
+			logutils.ZapLogger().Warn("failed to get addresses of network interface",
+				zap.String("networkInterface", ni.Name),
+				zap.Error(err))
+			continue
+		}
+
+		for _, addr := range addrs {
+			var ip net.IP
+			if ipNet := addrToIPNet(addr); ipNet == nil {
+				continue
+			} else {
+				ip = ipNet.IP
+			}
+			niIps = append(niIps, ip)
+		}
+
+		if len(niIps) > 0 {
+			ips = append(ips, niIps)
+		}
+	}
+
+	return ips, nil
+}
+
+// GetLocalAddressesForPairingServer is a high-level func
+// that returns a list of addresses to be used by local pairing server.
+func GetLocalAddressesForPairingServer() ([]net.IP, error) {
```

Comment:
It seems we can rename some functions for clarity:

GetLocalAddresses -> getLocalAddresses
FilterAddressesForPairingServer -> filterAddressesForPairingServer
Using lowercase function names indicates which functions are intended for external usage, while uppercase names imply internal-only functions. This naming convention would help differentiate between public and private methods at a glance.

Reply:
Thanks for this! I was actually planning to do this, but left this for the end of implementation. And then forgot to update the naming 😞